src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp

```diff
@@ -700,14 +700,17 @@ void prepare_primitive_fusing::fuse_simple_primitives(program &p) {
             if (_lo.get_optimization_attributes().use_onednn_impls) {
                 if (input.is_type<reshape>() || input.is_type<concatenation>())
                     return;
-                #ifdef ENABLE_ONEDNN_FOR_GPU
-                // Activation should not fused if it isn't supported in onednn
-                try {
-                    onednn::convert_activation_func(activation_node.get_primitive()->activation_function);
-                } catch (...) {
-                    return;
+
+                // Activation should not be fused if oneDNN does NOT support it
+                if (_lo.are_primitives_implemented_for_onednn(input) && p.get_engine().get_device_info().supports_immad)  {
```

Comment:
* `are_primitives` -> `is_primitive`
* no need to check `supports_immad` because we already checked `use_onednn_impls`